docs/source/operators/deploy-kubernetes.md

```diff
@@ -15,248 +17,495 @@ The following sample kernel specifications apply to Kubernetes deployments:
+   --kube-context [mycluster-context-name] \
+   --namespace [namespace-name]
+
 ```
 
-From anywhere with helm cluster access, create the service and deployment by running helm from a source release or the git repository:
+Alternatively, helm chart tarball is also accessible as an asset on our [release](https://github.com/jupyter-server/enterprise_gateway/releases) page, replace [VERSION] with specific release version you wantr to use:
+
+```bash
+helm  upgrade --install  enterprise-gateway \
+  https://github.com/jupyter-server/enterprise_gateway/releases/download/[VERSION]/jupyter_enterprise_gateway_helm-[VERSION].tgz \
+   --kube-context [mycluster-context-name] \
+   --namespace [namespace-name]
+```
+
+### Access to Enterprise Gateway from outside the cluster
+
+Take a look at the Kubernetes [documentation](https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster-services/#ways-to-connect) on how you can access the Kubernetes service from outside the cluster.
+
+A Kubernetes Ingress is the most user-friendly way of interacting with the service and that is what we will cover in this section.
+If you do not have a Kubernetes Ingress configured on your cluster the easiest way to get access will be using the NodePort service.
+
+#### Kubernetes Ingress Setup
+
+##### Prerequisites
+
+- Ingress controller deployed on your Kubernetes cluster. Review the Kubernetes [documentation](https://kubernetes.io/docs/concepts/services-networking/ingress/) for available options.
+- Wildcard DNS record is configured to point to the IP of the LoadBalancer, which frontends your ingress controller
+- Review specific Ingress controller configuration to enable wildcard path support if you are using Kubernetes version < v1.18
+- With Kubernetes v1.18 Ingress uses `PathType` parameter which is set to `Prefix` in the helm chart by default, so no additional configuration is required
+- Refer to your ingress controller documentation on how to setup TLS with your ingress
+
+##### Update Helm deployment to enable ingress
+
+Create file `values-ingress.yaml` with the following content:
 
 ```bash
-helm upgrade --install --namespace enterprise-gateway enterprise-gateway etc/kubernetes/helm/enterprise-gateway
+ingress:
+  enabled: true
+  # Ingress resource host
+  hostName: "[unique-fully-qualified-domanin-name]"
+
 ```
 
-the helm chart tarball is also accessible as an asset on our [release](https://github.com/jupyter-server/enterprise_gateway/releases) page:
+Add this file to your helm command and apply to the cluster replacing [PLACEHOLDER] with appropriate values for your environment:
 
 ```bash
-helm install --namespace enterprise-gateway enterprise-gateway https://github.com/jupyter-server/enterprise_gateway/releases/download/v2.6.0/jupyter_enterprise_gateway_helm-2.6.0.tgz
+helm  upgrade --install  jupyter-e-gw \
+  jupyter-server/enterprise_gateway/etc/kubernetes/helm/enterprise-gateway \
```

Comment:
Since we used `enterprise-gateway` in the very first (and default) example, I think we should be consistent elsewhere.  Also, the path should be relative to the repository again.
```suggestion
helm  upgrade --install enterprise-gateway \
  etc/kubernetes/helm/enterprise-gateway \
```